components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2022,10 +2026,11 @@ private void validateAndSetTransports(API api) throws APIManagementException {
      *
      * @param api Relevant API that need to be validated.
      */
-    private void validateAndSetAPISecurity(API api) {
+    private void validateAndSetAPISecurity(API api, APIProduct apiProduct) {
```

Comment:
As suggested previously lets add a separate overloaded function for APIProduct

Reply:
+1

Reply:
Fixed in 32e0ed7cffd83e1e1bfb9eaf3eed84e0bb15a83a